src/lib/code-sample/schema.ts

```diff
@@ -0,0 +1,58 @@
+import { z } from 'zod'
+
+import {
+  createJavascriptRequest,
+  createJavascriptResponse,
+} from './javascript.js'
+
+export const CodeSampleDefinitionSchema = z.object({
+  title: z.string().trim().min(1),
+  description: z.string().trim().min(1),
+  request: z.object({
+    path: z
+      .string()
+      .startsWith('/')
+      .regex(
+        /^[a-z_/]+$/,
+        'Can only contain the lowercase letters a-z, underscores, and forward slashes.',
+      ),
+    parameters: z.record(z.string().min(1), z.any()),
+  }),
+  response: z
+    .object({
+      body: z.record(z.string().min(1), z.any()),
+    })
```

Comment:
Is there a better type than `z.any`? We basically need to allow any JSON serializable type here.

Reply:
https://zod.dev/?id=json-type 
I think the zod recommended approach is to do something like this:
```
const literalSchema = z.union([z.string(), z.number(), z.boolean(), z.null()])
type Literal = z.infer<typeof literalSchema>
type Json = Literal | { [key: string]: Json } | Json[]

const jsonSchema: z.ZodType<Json> = z.lazy(() => 
  z.union([
    literalSchema,
    z.array(jsonSchema),
    z.record(jsonSchema)
  ])
)
```
and then change parameters to 
` 
   parameters: z.record(z.string().min(1), jsonSchema)`
and response to:
```
response: z
    .object({
      body: z.record(z.string().min(1), jsonSchema),
    })
})